File: python/factorial.py
# factorial of a number using recursion
def fact(n):
  if n==1:
    return n
  else:
    return n*fact(n-1)
number = int(input())
if number<0:
  print("doesn't exist for negative number")
elif number==0:
  print("factorial of 0 is 1")
else:
  print("factorial of",number,"is",fact(number))

File: c/factorial.c
//program to find the factorial of the number
#include<stdio.h>
#include<conio.h>
main()
{
    int n;
    long ans;
    printf("enter a number");
    scanf("%d",&n);
    ans=fact(n);
    printf("factorail of %d is %ld",n,ans);
 }
 
 long fact(int n)
 {
    if(n==1)
    {
       return(1);
    }
    else
    {
       if(n>1)
       {
          return(n*fact(n-1));
       }
    }
 }
